server/protocols/jmap-rfc-8621/src/main/scala/org/apache/james/jmap/method/EmailGetMethod.scala

```diff
@@ -69,17 +69,36 @@ class EmailGetMethod @Inject() (serializer: Serializer,
   override def process(capabilities: Set[CapabilityIdentifier], invocation: Invocation, mailboxSession: MailboxSession, processingContext: ProcessingContext): Publisher[(Invocation, ProcessingContext)] =
     metricFactory.decoratePublisherWithTimerMetricLogP99(JMAP_RFC8621_PREFIX + methodName.value,
       asEmailGetRequest(invocation.arguments)
-        .flatMap(request => getEmails(request, mailboxSession))
-        .map(response => Invocation(
-          methodName = methodName,
-          arguments = Arguments(serializer.serialize(response).as[JsObject]),
-          methodCallId = invocation.methodCallId))
+        .flatMap(computeResponseInvocation(_, invocation, mailboxSession))
         .onErrorResume({
           case e: IllegalArgumentException => SMono.just(Invocation.error(ErrorCode.InvalidArguments, e.getMessage, invocation.methodCallId))
           case e: Throwable => SMono.raiseError(e)
         })
         .map(invocationResult => (invocationResult, processingContext)))
 
+  private def computeResponseInvocation(request: EmailGetRequest, invocation: Invocation, mailboxSession: MailboxSession): SMono[Invocation] =
+    validateProperties(request)
+        .fold(
+          e => SMono.raiseError(e),
```

Comment:
no need to return to the line here?

Reply:
I did prefer having the two case handled at the same level, it makes it visually more clear.